iso15118impls/slac_session.py

```diff
@@ -77,27 +85,91 @@ async def start(self, cs_config: dict):
+    """
+    Set the EVSE key for the session.
+    This function is adapted to work with the ChargebyteChargingStation.
+    """
+    from binascii import unhexlify
+    from pyslac.layer_2_headers import EthernetHeader, HomePlugHeader
+    from pyslac.messages import SetKeyReq, SetKeyCnf
+    from pyslac.enums import FramesSizes, Timers, MMTYPE_REQ, CM_SET_KEY, SLAC_SETTLE_TIME
+
+    logger.info("CM_SET_KEY: Started...")
+    # Set NMK and NID with the environment variables NMK and NID 
+    nmk = unhexlify(os.environ.get("NMK"))
+    nid = unhexlify(os.environ.get("NID"))
+    logger.debug("New NMK: %s", os.environ.get("NMK"))
+    logger.debug("New NID: %s", os.environ.get("NID"))
+    ethernet_header = EthernetHeader(
+        dst_mac=self.evse_plc_mac, src_mac=self.evse_mac
+    )
+    homeplug_header = HomePlugHeader(CM_SET_KEY | MMTYPE_REQ)
+    key_req_payload = SetKeyReq(nid=nid, new_key=nmk)
+
+    frame_to_send = (
+        ethernet_header.pack_big()
+        + homeplug_header.pack_big()
+        + key_req_payload.pack_big()
+    )
+
+    # TODO: Change this to just open a socket once for every SlacSession
+    # and not every time we call send or send_recv_eth
+    # Also think about including the send, rcv method as inner methods of
+    # SetKeyReq. Maybe even create a class SetKey that handles both the
+    # Send and the CNF of the message
+    try:
+        await self.send_frame(frame_to_send)
+        data_rcvd = await self.rcv_frame(
+            rcv_frame_size=FramesSizes.CM_SET_KEY_CNF,
+            timeout=Timers.SLAC_INIT_TIMEOUT,
+        )
+    except asyncio.TimeoutError as e:
+        raise TimeoutError("SetKey Timeout raised") from e
+    try:
+        SetKeyCnf.from_bytes(data_rcvd)
+        self.nmk = nmk
+        self.nid = nid
+    except ValueError as e:
+        logger.error(e)
+        if self.nmk and self.nid:
+            logger.debug(
+                "SetKeyReq has failed, old NMK: %s and NID: %s apply",
+                self.nmk,
+                self.nid,
+            )
+        else:
+            raise ValueError("SetKeyCnf data parsing into the class failed") from e
+    logger.debug("Registering NMK and NID into the PLC node...")
+    await asyncio.sleep(SLAC_SETTLE_TIME)
+    logger.info("CM_SET_KEY: Finished!")
+    return data_rcvd
+
+def get_slac_handler(low_level_abstraction: ChargingStation, env_path: Optional[str] = None) -> SlacHandler:
     """
     Factory function to create a SlacHandler instance with the given low-level abstraction.
     """
     slac_config = Config()
     slac_config.load_envs(env_path)
 
-    return SlacHandler(slac_config, _low_level_abstraction)
+    return SlacHandler(slac_config, low_level_abstraction)
 
-def get_cs_config() -> dict:
+def get_cs_config(network_interface) -> dict:
     """
     Factory function to get the charging station configuration.
     """
+    CONFIG_PATH = Path(__file__).parent / "cs_configuration.json"
```

Comment:
I am starting to realize: This config file is only used in your code and not passed to any of the pyslac code, right?
I assume you copied the idea of having a config file from the pyslac examples?
Maybe we can get rid of this JSON file altogether?
We will always only have one EVSE. We can have a default hardcoded EVSE ID, but allow to modify it using env variables. The network interface is also already configurable.